system/inc/system_cloud.h

```diff
@@ -206,7 +199,28 @@ typedef struct {
  * @note The periodic functionality is not available for the Spark Core.
  */
 int spark_publish_vitals(system_tick_t period_s, void *reserved);
-bool spark_send_event(const char* name, const char* data, int ttl, uint32_t flags, void* reserved);
+
+typedef spark_protocol_send_event_data spark_send_event_data;
```

Comment:
I'd keep these structures separate to make it clear that `spark_send_event()` and `spark_protocol_send_event()` can evolve independently, and we're not necessarily going to expose all protocol features to the application layer.

Reply:
I agree, and that's fine they can diverge later. if they do need to diverge then we can split them out as separate types and add the requisite copying, but for now it's more efficient to use a typedef, since they are the same thing at present.  Let's not take the hit now for something we may not need.